Commonalities/documentation/Working/user-consent.md

```diff
@@ -0,0 +1,396 @@
+# User Consent Management
+
+This document captures guidelines, compliant with GDPR requirements, for Operator NaaS platform to handle user consents.
+
+## Table of Contents
+
+  - [Introduction](#introduction)
+  - [Glossary of Terms and Concepts](#glossary-of-terms-and-concepts)
+  - [Purposes](#purposes)
+    - [Purpose levels](#purpose-levels)
+    - [Purpose relationship with other key concepts](#purpose-relationship-with-other-key-concepts)
+  - [Login authorization requirements and purpose usage](#login-authorization-requirements-and-purpose-usage)
+    - [Using purpose parameter in the authorization request](#using-purpose-parameter-in-the-authorization-request)
+    - [Non-automatic purpose requirements for the Login authorization](#non-automatic-purpose-requirements-for-the-login-authorization)
+  - [Obtaining an access token for non-automatic purposes](#obtaining-an-access-token-for-non-automatic-purposes)
+  - [Consent API and GDPR API](#consent-api-and-gdpr-api)
+  - [Annex: Flow source code](#annex-flow-source-code)
+    - [Delegated authorization jwt-bearer flow](#delegated-authorization-jwt-bearer-flow)
+    - [Consent gathering with Consent API flow](#consent-gathering-with-consent-api-flow)
+
+
+## Introduction
+
+Operator NaaS platform implementing CAMARA should be built following a Privacy By Design approach to fully comply with the spirit and letter of the [GDPR regulation](https://gdpr-info.eu/), which provides a great level of protection for user privacy. This means that an API that process Personal Information needs a user consent. Consents are given by users to legal entities to process personal data under a specific purpose.
+
+This document captures guidelines for Operator NaaS platform to handle user consents to comply with GDPR requirements in an easy way. It introduces the concept of **purpose** in OpenID Connect, which is built on top of OAuth 2.0. These User consent management guidelines define a procedure for application back-ends to obtain access tokens for purposes which require user consent. It relies on an existing previous authorization session of the user (where user identity was validated against the corresponding Identity Provider). Application back-ends will be then able to get access tokens with the required purposes on behalf of the user (i) referring to a previous authorization, and (ii) without requiring further user interaction in between, once the user consent is collected. 
+
```

Comment:
Is there an intent to make use of RFC8693? https://www.rfc-editor.org/rfc/rfc8693.html

Reply:
Not for the moment. In fact, we plan to update the document to provide a solution based on ["OpenID Connect Client-Initiated Backchannel Authentication (CIBA) Flow"](https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html). We are going to present it Today in CAMARA meeting. CIBA standard is based on back-end to back-end flow which allows to meet scenarios with IoT devices, scenarios without App interaction, scenarios with or without user authorization (for trusted partners)... this would meet the requirements that we have been gathering from partners and operators.

Reply:
In any case, do you see any potential scenario where RFC8693 is required? If so, which one?